tests/components/blue_current/__init__.py

```diff
@@ -1,52 +1,94 @@
 """Tests for the Blue Current integration."""
 from __future__ import annotations
 
-from unittest.mock import patch
+from asyncio import Event, Future
+from unittest.mock import MagicMock, patch
 
 from bluecurrent_api import Client
 
-from homeassistant.components.blue_current import DOMAIN, Connector
-from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import async_dispatcher_send
 
 from tests.common import MockConfigEntry
 
+DEFAULT_CHARGE_POINT = {
+    "evse_id": "101",
+    "model_type": "",
+    "name": "",
+}
+
+
+def create_client_mock(
+    loop_future: Future | None,
+    charge_point: dict,
+    status: dict | None,
+    grid: dict | None,
+) -> MagicMock:
+    """Create a mock of the bluecurrent-api Client."""
+    client_mock = MagicMock(spec=Client)
+    client_mock.started_loop = Event()
+
+    async def start_loop(receiver):
+        """Set the receiver and await future."""
+        client_mock.receiver = receiver
+
+        client_mock.started_loop.set()
+        client_mock.started_loop.clear()
+
+        if loop_future and not loop_future.done():
```

Comment:
Why is the future optional?

Reply:
Because the sensor tests don't need it. I could still pass the fixture but isn't this cleaner?

Reply:
Don't we always want to make sure the background task is alive to test realistically?

Reply:
Moved the future creation to `init_integration` and the future is no longer optional. 

One thing is that to make it more realistic, the future should be recreated every time `start_loop` is called. Right now it just keeps throwing the exception set in a test. I don't really know how to do that and keep it accessible (in tests) other and creating a class that encapsulates the Future or something?

Reply:
I'd define a pytest fixture that returns a dict or a dataclass that can store the future. Create the future in the `start_loop` side effect and then store the future in the dict or dataclass. Then the tests can access the future via the fixture.